### ISSUE TITLE
iOS permission dialog

### ISSUE THREAD

**Issue by gassaki** [fechanique/cordova-plugin-fcm#220]
The permission dialog in iOS opens right when the app is launched. Is there a way to ask for permission just when I set it to?

**Comment by fechanique** [comment]
is not possible by the moment

**Comment by shi11** [comment]
Hmm, that's no good from a UX standpoint. I need a solution to suppress it as I control it using the cordova-dialogs plugin

**Comment by Llermy** [comment]
@shi11 i agree with you, i don't think any user would accept any authorization without an explanation or some info before. So I've added a new function FCMPlugin.ready() that wraps all the code to register for notifications and show the pop-up in iOS. It should be called before setting the notifications with FCMPlugin.onNotification(). I've just done a pull request to add that functionality ;) It would be great if someone with experience in c obj / iOS could review the code, it's my first time writing for iOS. :sweat_smile:

**Comment by shi11** [comment]
@Llermy Great work. I reviewed the code and it makes sense but I won't be able to test it at the moment. I ended up implementing the cordova-plugin-firebase plugin

**Comment by nivaBurleigh** [comment]
Is it possible now to have this functionality without manually changing the ios fcm plugin?
